src/services/advancedSearchQuery/visitors/QueryBuilderVisitor.php

```diff
@@ -0,0 +1,118 @@
+<?php declare(strict_types=1);
+/**
+ * @author Nicolas CARPi <nico-git@deltablot.email>
+ * @author Marcel Bolten <github@marcelbolten.de>
+ * @copyright 2012 Nicolas CARPi
+ * @see https://www.elabftw.net Official website
+ * @license AGPL-3.0
+ * @package elabftw
+ */
+
+namespace Elabftw\Services\AdvancedSearchQuery\Visitors;
+
+use function array_merge;
+use function bin2hex;
+use Elabftw\Services\AdvancedSearchQuery\Collectors\WhereCollector;
+use Elabftw\Services\AdvancedSearchQuery\Grammar\AndExpression;
+use Elabftw\Services\AdvancedSearchQuery\Grammar\AndOperand;
+use Elabftw\Services\AdvancedSearchQuery\Grammar\NotExpression;
+use Elabftw\Services\AdvancedSearchQuery\Grammar\OrExpression;
+use Elabftw\Services\AdvancedSearchQuery\Grammar\OrOperand;
+use Elabftw\Services\AdvancedSearchQuery\Grammar\SimpleValueWrapper;
+use Elabftw\Services\AdvancedSearchQuery\Interfaces\Visitable;
+use Elabftw\Services\AdvancedSearchQuery\Interfaces\Visitor;
+use PDO;
+use function random_bytes;
+
+class QueryBuilderVisitor implements Visitor
+{
+    public function buildWhere(Visitable $parsedQuery, string $column): WhereCollector
+    {
+        return $parsedQuery->accept($this, new VisitorParameters($column));
+    }
+
+    public function visitSimpleValueWrapper(SimpleValueWrapper $simpleValueWrapper, VisitorParameters $parameters): WhereCollector
+    {
+        $value = $simpleValueWrapper->getValue();
+        $param = ':' . bin2hex(random_bytes(5));
```

Comment:
what's the purpose of this random stuff? Is it there to have a unique identifier for the placeholder in the query?

Reply:
Yes, that is exactly what it does.
Is this a bad idea?

Reply:
ok I understand. Then yes add a comment to explain the why.